### DISCUSSION TITLE
Recovery password flow with API

### DISCUSSION THREAD

**Discussion by Kostanos** [ory/kratos#1388] [Q&A] [answered]
Hi,
we can't find a way to finalize the Password Recovery flow with API.
1. Get recovery form. Status: **works**
_By using the API call as in documentation._
2. Create recovery flow and send password recovery email to the user. Status: **works**
_User gets the email with the link to Kratos public endpoint._
3. Get the password change form from kratos by API. Status: **not working**
_User clicks to the link
Browser opens the Kratos and redirects to the Redirect URL
The Redirect URL is handled by Mobile APP (universal link)_
**How, mobile app will get the password change form, if this form API call requires to have API bearer token?**

Any advice please?

**Comment by aeneasr** [comment, marked as answer]
Recovery with  pure API is currently not implemented, see: https://www.ory.sh/kratos/docs/next/self-service/flows/account-recovery/#send-recovery-link-to-email

We are planning however to move from link confirmation to code confirmation for recovery and verification flows, which should make this possible in the future!
